scalafix-core/shared/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -71,20 +76,61 @@ case class ExplicitResultTypes(
     case _: Defn.Var => MemberKind.Var
   }
 
+  @tailrec private def isOwner(owner: Symbol, symbol: Symbol): Boolean =
+    if (symbol == owner) true
+    else {
+      symbol match {
+        case Symbol.None => false
+        case Symbol.Global(qual, _) => isOwner(owner, qual)
+        case _ => false
+      }
+    }
+
   override def fix(ctx: RuleCtx): Patch = {
+    val table = index.asInstanceOf[EagerInMemorySemanticdbIndex]
+    val shorten =
+      if (config.unsafeShortenNames) Shorten.Readable
+      else Shorten.FullyQualified
     def defnType(defn: Defn): Option[(Type, Patch)] =
       for {
         name <- defnName(defn)
-        denot <- name.denotation
-        tpe <- denot.tpeInternal
-        // Skip existential types for now since they cause problems.
-        if !tpe.tag.isExistentialType
-        result <- TypeSyntax.prettify(
-          tpe,
-          ctx,
-          config.unsafeShortenNames,
-          name.pos)
-      } yield result
+        defnSymbol <- name.symbol
+        info <- table.info(defnSymbol.syntax)
+        result <- {
+          try {
+            import scala.meta.internal.{semanticdb3 => s}
+            val tpe = info.tpe.get.tag match {
+              case s.Type.Tag.METHOD_TYPE =>
+                info.tpe.get.methodType.get.returnType.get
+              case _ => info.tpe.get
+            }
+            Some(PrettyType.toType(tpe, table, shorten))
+          } catch {
```

Comment:
minor, but for readability I would extract the error handling code. Something like:

```scala
result <- withErrorReporting {
  import scala.meta.internal.{semanticdb3 => s}
  val tpe = info.tpe.get.tag match {
    case s.Type.Tag.METHOD_TYPE =>
      info.tpe.get.methodType.get.returnType.get
    case _ => info.tpe.get
  }
  Some(PrettyType.toType(tpe, table, shorten))
}
```

Reply:
I extracted the logic into a separate method. Error handling needs quite a bit of context so it's difficult to implement a generic `withErrorReporting` utility